rotkehlchen/exchanges/bitstamp.py

```diff
@@ -0,0 +1,715 @@
+        deserialize it.
+
+        Can raise DeserializationError.
+
+        From Bitstamp documentation, deposits/withdrawals can have a fee
+        (the amount is expected to be in the currency involved)
+        https://www.bitstamp.net/fee-schedule/
+
+        Bitstamp support confirmed the following withdrawal JSON:
+        {
+            "fee": "0.00050000",
+            "btc_usd": "0.00",
+            "datetime": "2020-12-04 09:30:00.000000",
+            "usd": "0.0",
+            "btc": "-0.50000000",
+            "type": "1",
+            "id": 123456789,
+            "eur": "0.0"
+        }
+        NB: any asset key not related with the pair is discarded (e.g. 'eur').
+        """
+        type_ = raw_movement['type']
+        category: AssetMovementCategory
+        if type_ == 0:
+            category = AssetMovementCategory.DEPOSIT
+        elif type_ == 1:
+            category = AssetMovementCategory.WITHDRAWAL
+        else:
+            raise AssertionError(f'Unexpected Bitstamp asset movement case: {type_}.')
+
+        timestamp = iso8601ts_to_timestamp(raw_movement['datetime'])
+        trade_pair_data = self._get_trade_pair_data_from_transaction(raw_movement)
+        base_asset_amount = deserialize_asset_amount(
+            raw_movement[trade_pair_data.base_asset_symbol],
+        )
+        quote_asset_amount = deserialize_asset_amount(
+            raw_movement[trade_pair_data.quote_asset_symbol],
+        )
+        amount: FVal
+        fee_asset: Asset
+        if base_asset_amount != ZERO and quote_asset_amount == ZERO:
+            amount = base_asset_amount
+            fee_asset = trade_pair_data.base_asset
+        elif base_asset_amount == ZERO and quote_asset_amount != ZERO:
+            amount = quote_asset_amount
+            fee_asset = trade_pair_data.quote_asset
+        else:
+            raise DeserializationError(
+                'Could not deserialize Bitstamp asset movement from user transaction. '
+                f'Unexpected asset amount combination found in: {raw_movement}.',
+            )
+
+        asset_movement = AssetMovement(
+            timestamp=timestamp,
+            location=Location.BITSTAMP,
+            category=category,
+            address=None,  # requires query "crypto_transactions" endpoint
+            transaction_id=None,  # requires query "crypto_transactions" endpoint
+            asset=fee_asset,
+            amount=abs(amount),
+            fee_asset=fee_asset,
+            fee=deserialize_fee(raw_movement['fee']),
+            link=str(raw_movement['id']),
+        )
+        return asset_movement
+
+    def _deserialize_trade(
+            self,
+            raw_trade: Dict[str, Any],
+    ) -> Trade:
+        """Process a trade user transaction from Bitstamp and deserialize it.
+
+        Can raise DeserializationError.
+        """
+        timestamp = iso8601ts_to_timestamp(raw_trade['datetime'])
```

Comment:
This is actually a mistake. I realized it only thanks to the test you yourself wrote @vnavascues. (It fails on my system since I am not in UTC).

`iso8601ts_to_timestamp()` interprets the datetime string to be in local time. We had similar problem with poloniex. So I am pushing a fix for this before merging.